src/CMakeLists.txt

```diff
@@ -88,7 +88,7 @@ set_target_properties(gpuarray PROPERTIES
   INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib
   MACOSX_RPATH OFF
   # This is the shared library version
-  VERSION 0.0
+  VERSION 0.1
```

Comment:
I don't think that it is a problem that changing the minor version here will not trigger a recompilation.  Everything that worked with 0.0 will work with 0.1.  That is the point of this version scheme.

If we were to change the major version it would make the currently compiled modules unloadable, possibly triggering a recompilation (I'm not 100% sure how Theano deal with C code it can't reload).

If people update Theano and it uses newly introduced symbols, they will need to update libgpuarray also, yes.  Usually this is handled with recommended versions for releases and a guideline of the style: use the latest master of libgpuarray for the latest master of Theano.

My problem with maintaining another version scheme is that it duplicates works and we will forget to bump one or the other for some changes, which will lead to exactly the problems that you describe, except that nobody will expect them.